src/simplewallet/simplewallet.cpp

```diff
@@ -602,12 +616,14 @@ std::string simple_wallet::get_command_usage(const std::vector<std::string> &arg
 
 bool simple_wallet::viewkey(const std::vector<std::string> &args/* = std::vector<std::string>()*/)
 {
-  if (m_wallet->ask_password() && !get_and_verify_password()) { return true; }
+  SCOPED_WALLET_UNLOCK();
```

Comment:
I don't think this will make any difference because the view key doesn't get encrypted.